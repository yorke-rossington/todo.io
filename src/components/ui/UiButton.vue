<template>
    <button
        type="submit"
        class="mt-3 flex w-full items-center justify-center rounded-md bg-indigo-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 sm:mt-0 sm:w-auto"
    >
        <slot></slot>
    </button>
</template>

<script setup lang="ts"></script>
