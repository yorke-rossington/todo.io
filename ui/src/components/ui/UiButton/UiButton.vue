<template>
    <button
        class="flex items-center justify-center transition text-base font-semibold shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
        :class="[
            props.variant === 'primary'
                ? 'bg-indigo-600 hover:bg-indigo-500 text-white'
                : '',
            props.variant === 'secondary'
                ? 'bg-gray-200 hover:bg-gray-100 text-gray-800'
                : '',
            props.circular ? 'rounded-full' : 'rounded-md',
            props.icon ? 'p-1' : 'px-3 py-2'
        ]"
    >
        <slot></slot>
    </button>
</template>

<script setup lang="ts">
type UiButtonProps = {
    variant?: "primary" | "secondary";
    circular?: boolean;
    icon?: boolean;
};

const props = withDefaults(defineProps<UiButtonProps>(), {
    variant: "primary",
    circular: false
});
</script>
