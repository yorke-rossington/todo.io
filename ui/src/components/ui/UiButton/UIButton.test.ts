import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import UiButton from "./UiButton.vue";

describe("UIButton", () => {
    it("Renders primary button", () => {
        const comp = mount(UiButton, {
            props: { variant: "primary" },
            slots: { default: "Test Button" }
        });

        expect(comp.element).toMatchSnapshot();
    });

    it("Renders secondary button", () => {
        const comp = mount(UiButton, {
            props: { variant: "secondary" },
            slots: { default: "Test Button" }
        });

        expect(comp.element).toMatchSnapshot();
    });

    it("Renders circular button", () => {
        const comp = mount(UiButton, {
            props: { circular: true },
            slots: { default: "Test Button" }
        });

        expect(comp.element).toMatchSnapshot();
    });

    it("Renders icon button", () => {
        const comp = mount(UiButton, {
            props: { icon: true },
            slots: { default: "Test Button" }
        });

        expect(comp.element).toMatchSnapshot();
    });

    it("Renders the correct slot content", () => {
        const comp = mount(UiButton, {
            props: { icon: true },
            slots: { default: "A Great Button" }
        });

        expect(comp.text()).toBe("A Great Button");
        expect(comp.element).toMatchSnapshot();
    });
});
