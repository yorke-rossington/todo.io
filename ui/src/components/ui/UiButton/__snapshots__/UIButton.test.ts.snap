// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UIButton > Renders circular button 1`] = `
<button
  class="flex items-center justify-center transition text-base font-semibold shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 bg-indigo-600 hover:bg-indigo-500 text-white rounded-full px-3 py-2"
>
  
  Test Button
  
</button>
`;

exports[`UIButton > Renders icon button 1`] = `
<button
  class="flex items-center justify-center transition text-base font-semibold shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 bg-indigo-600 hover:bg-indigo-500 text-white rounded-md p-1"
>
  
  Test Button
  
</button>
`;

exports[`UIButton > Renders primary button 1`] = `
<button
  class="flex items-center justify-center transition text-base font-semibold shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 bg-indigo-600 hover:bg-indigo-500 text-white rounded-md px-3 py-2"
>
  
  Test Button
  
</button>
`;

exports[`UIButton > Renders secondary button 1`] = `
<button
  class="flex items-center justify-center transition text-base font-semibold shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 bg-gray-200 hover:bg-gray-100 text-gray-800 rounded-md px-3 py-2"
>
  
  Test Button
  
</button>
`;

exports[`UIButton > Renders the correct slot content 1`] = `
<button
  class="flex items-center justify-center transition text-base font-semibold shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 bg-indigo-600 hover:bg-indigo-500 text-white rounded-md p-1"
>
  
  A Great Button
  
</button>
`;
